File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/termo/Constante.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  javax.persistence.Column
 *  javax.persistence.Embeddable
 *  org.hibernate.validator.NotNull
 */
package br.jus.trt8.pjecalc.negocio.dominio.termo;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.comum.rotinasdecalculo.CalculoDoProporcionalizar;
import br.jus.trt8.pjecalc.negocio.dominio.termo.ParametroDoTermo;
import br.jus.trt8.pjecalc.negocio.dominio.termo.Termo;
import java.math.BigDecimal;
import javax.persistence.Column;
import javax.persistence.Embeddable;
import org.hibernate.validator.NotNull;

@Embeddable
public class Constante
implements Termo {
    private static final long serialVersionUID = -7993361365356987338L;
    @NotNull
    @Column(name="RVLVALOR", precision=38, scale=25)
    private BigDecimal valor;

    @Override
    public BigDecimal resolverValor(ParametroDoTermo parametro) {
        if (parametro.getVerbaDeCalculo().getAplicarProporcionalidade().booleanValue() && this.getValor() != null) {
            int diasParaExcluir = 0;
            if (parametro.getVerbaDeCalculo().getExcluirFeriasGozadas().booleanValue()) {
                diasParaExcluir += parametro.getCalculo().obterDiasFerias(parametro.getPeriodo());
            }
            if (parametro.getPeriodo().totalDeDias() - diasParaExcluir == 31) {
                diasParaExcluir = 1;
            }
            if (parametro.getVerbaDeCalculo().getExcluirFaltaJustificada().booleanValue()) {
                diasParaExcluir += parametro.getCalculo().obterFaltasJustificadas(parametro.getPeriodo());
            }
            if (parametro.getVerbaDeCalculo().getExcluirFaltaNaoJustificada().booleanValue()) {
                diasParaExcluir += parametro.getCalculo().obterFaltasNaoJustificadas(parametro.getPeriodo());
            }
            parametro.setValorIntegral(this.valor);
            CalculoDoProporcionalizar calculoDoProporcionalizar = new CalculoDoProporcionalizar(parametro.getPeriodo(), this.getValor(), diasParaExcluir);
            calculoDoProporcionalizar.executar();
            return calculoDoProporcionalizar.getResultado();
        }
        return this.getValor();
    }

    public BigDecimal getValor() {
        return this.valor;
    }

    public void setValor(BigDecimal valor) {
        this.valor = valor;
    }

    public String toString() {
        return Utils.formatarNumero(this.valor);
    }
}

File: src/lib/pjecalc/core/dominio/termo/constante.ts
/**
 * Porte 1:1 de Constante.java (64 linhas).
 * Termo que devolve um valor fixo, opcionalmente proporcionalizado.
 *
 * Ref: pjecalc-fonte/.../dominio/termo/Constante.java:28-50
 */
import Decimal from 'decimal.js';
import type { Termo } from './termo';
import type { ParametroDoTermo } from './parametro-do-termo';
import { CalculoDoProporcionalizar } from '../../comum/rotinasdecalculo/calculo-do-proporcionalizar';

export class Constante implements Termo {
  private valor: Decimal | null = null;

  resolverValor(parametro: ParametroDoTermo): Decimal {
    const verba = parametro.getVerbaDeCalculo() as unknown as {
      getAplicarProporcionalidade?: () => boolean;
      getExcluirFeriasGozadas?: () => boolean;
      getExcluirFaltaJustificada?: () => boolean;
      getExcluirFaltaNaoJustificada?: () => boolean;
    };
    const calculo = parametro.getCalculo() as unknown as {
      obterDiasFerias?: (p: unknown) => number;
      obterFaltasJustificadas?: (p: unknown) => number;
      obterFaltasNaoJustificadas?: (p: unknown) => number;
    };
    const periodo = parametro.getPeriodo();
    const aplicarProp = verba.getAplicarProporcionalidade?.() ?? false;

    if (aplicarProp && this.valor !== null && periodo) {
      let diasParaExcluir = 0;
      if (verba.getExcluirFeriasGozadas?.()) {
        diasParaExcluir += calculo.obterDiasFerias?.(periodo) ?? 0;
      }
      // Java Constante.java:35-37: ajuste para mês de 31 dias
      const totalDeDias = (periodo as { totalDeDias?: () => number }).totalDeDias?.() ?? 30;
      if (totalDeDias - diasParaExcluir === 31) {
        diasParaExcluir = 1;
      }
      if (verba.getExcluirFaltaJustificada?.()) {
        diasParaExcluir += calculo.obterFaltasJustificadas?.(periodo) ?? 0;
      }
      if (verba.getExcluirFaltaNaoJustificada?.()) {
        diasParaExcluir += calculo.obterFaltasNaoJustificadas?.(periodo) ?? 0;
      }
      parametro.setValorIntegral(this.valor);
      const calc = new CalculoDoProporcionalizar(periodo as unknown as never, this.valor, diasParaExcluir);
      calc.executar();
      return calc.getResultado();
    }
    return this.valor ?? new Decimal(0);
  }

  getValor(): Decimal | null { return this.valor; }
  setValor(v: Decimal | null): void { this.valor = v; }

  toString(): string { return this.valor?.toString() ?? '0'; }
}
